### PR TITLE
completed ticket 6

### PR DESCRIPTION
**Pull request by YoosufAhmad**
created a comment section in my single article

### CODE REVIEW

File: src/components/CommentsCard.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  useEffect(() => {
+    getComments(article_id)
+      .then((res) => {
+        console.log(res, 'response in useEffect')
+        setComments(res);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
```
Comment (xViolaine) [inline src/components/CommentsCard.jsx]: This console.log doesn't really help the user - maybe you could set a boolean state to indicate whether there's an error or not, and conditionally display a message to the user if there's an error?

File: src/components/CommentsCard.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { getComments } from "../utils/api";
+import { useEffect, useState } from "react";
+
+const CommentsCard = ({ article_id }) => {
+  const [comments, setComments] = useState([]);
+
+  useEffect(() => {
+    getComments(article_id)
+      .then((res) => {
+        console.log(res, 'response in useEffect')
```
Comment (xViolaine) [inline src/components/CommentsCard.jsx]: Make sure to get rid of stray console.logs!

File: src/utils/api.js [hunk 3]
```diff
@@ -19,8 +19,19 @@ export const getArticles = () => {
 export const getArticleById = (article_id) => {
     return newsApi
     .get(`/articles/${article_id}`)
+    .then(({data}) => {
+        return data.article
+    })
+    .catch((error) => {
```
Comment (xViolaine) [inline src/utils/api.js:25]: See comment above :)

File: src/components/CommentsCard.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { getComments } from "../utils/api";
+import { useEffect, useState } from "react";
+
+const CommentsCard = ({ article_id }) => {
+  const [comments, setComments] = useState([]);
+
+  useEffect(() => {
+    getComments(article_id)
+      .then((res) => {
```
Comment (xViolaine) [inline src/components/CommentsCard.jsx:12]: We should implement a loading state so the users know the comments are still loading

**Review by xViolaine — CHANGES_REQUESTED** [review]
Hey there, please make sure to change your error handling and implement a loading state for the comments.

Furthermore, you should display some kind of message to the user if there are no comments for the current article :)

**Review by mezzDavies — COMMENTED** [review]
The loading is sorted good work 👍 

However your error handling/catch blocks aren't working as you want. If you remove the catch blocks from your api call functions then the catch blocks in your components (attached to the promise chain after the invocation of one of these functions) will pick up the error and the functionality you want to do there will come into affect. Also I'm not sure the use of throwing errors is working - I would simply render a message on-screen instead.

**Pull request merged**